### PR TITLE
Add dep-approvers for reviewing go dependency changes

### PR DESCRIPTION
**Pull request by cblecker**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
Add dep-approvers for reviewing go dependency changes

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: OWNERS [hunk 1]
```diff
@@ -31,3 +31,11 @@ filters:
   "BUILD(\\.bazel)?$":
     approvers:
       - ixdy
+
+  # go.{mod,sum} files relate to go dependencies, and should be reviewed by the
+  # dep-approvers
+  "go\\.(mod|sum)$":
+    required_reviewers:
+      - kubernetes/dep-approvers
```
Comment (nikhita) [inline OWNERS]: Does this work for github teams?

Comment (cblecker) [reply OWNERS]: Yup :)

**Comment by nikhita** [comment]
I guess...
/sig release

/priority important-soon

File: OWNERS (hunk 1 above)
Comment (nikhita) [reply OWNERS]: Nice!

**Review by nikhita — APPROVED** [review]
/lgtm

/assign @liggitt 
for approval

**Pull request merged**